backend/controllers/userController.js

```diff
@@ -0,0 +1,66 @@
+const asyncHandler = require("express-async-handler")
+const generateToken = require("../utils/generateToken")
+const { ErrorHandler } = require("../middleware/errorMiddleware")
+const User = require("../model/user.model")
+
+//@desc Register a new user
+//@route POST /users/registerUser
+
+const registerUser = asyncHandler(async (req, res, next) => {
+  try {
+    const { name, email, password, confirmpwd, phone } = req.body
+
+    const userExists = await User.findOne({ email })
+
+    if (userExists) {
+      throw new ErrorHandler(403, "User already exists, try login")
+    }
+    if (password != confirmpwd) {
+      throw new ErrorHandler(403, "Passwords do not match")
+    }
+
+    const user = await User.create({
+      name,
+      email,
+      password,
+      phone,
+    })
+
+    if (user) {
+      res.status(201).json({
+        success: true,
+        user: user,
+        token: generateToken(user._id),
+        message: "User registered successfully",
+      })
+    }
+  } catch (err) {
+    next(err)
```

Comment:
This is what I was asking for having in the earlier PR. Can you conclude the benefits of it?

Reply:
it will return the error after the execution of all the middleware functions is completed

Reply:
Yes, but where will the compiler go after reaching 38 line?

Reply:
errorMiddleware